integration-tests/api/__tests__/store/cart.js

```diff
@@ -480,6 +487,60 @@ describe("/store/carts", () => {
       ])
     })
 
+    it("adding line item to a cart with associated sales channel returns 400", async () => {
+      const api = useApi()
+
+      const response = await api
+        .post(
+          "/store/carts/test-cart-with-sales-channel/line-items",
+          {
+            variant_id: "test-variant-quantity", // variant's product doesn't belong to a sales channel
+            validate_sales_channels: true,
+            quantity: 1,
+          },
+          { withCredentials: true }
+        )
+        .catch((err) => err.response)
+
+      expect(response.status).toEqual(400)
+      expect(response.data.type).toEqual("invalid_data")
+      expect(response.data.message).toEqual(
+        "Variant being added to the cart doesn't belong to current sales channel that is set on the cart."
```

Comment:
suggestion: `The product ${product.title} must belongs to the sales channel on which the cart has been created`